packages/pwa/app/utils/utils.js

```diff
@@ -158,7 +160,112 @@ export const capitalize = (text) => {
 }
 
 /**
- * A util to return current url configuration
- * @returns {object} - url object from the pwa-kit.config.json file
+ * This function return the identifiers (site and locale) from the given url
+ * The site will always go before locale if both of them are presented in the pathname
+ * @param path {string}
+ * @returns {{siteRef: string, localeRef: string}} - site and locale reference (it could either be id or alias)
  */
-export const getUrlConfig = () => getConfig().app.url
+export const getParamsFromPath = (path) => {
+    const {pathname, search} = new URL(absoluteUrl(path))
+
+    const config = getConfig()
+    const {pathMatcher, searchMatcherForSite, searchMatcherForLocale} = getConfigMatcher(config)
+    const pathMatch = pathname.match(pathMatcher)
+    const searchMatchForSite = search.match(searchMatcherForSite)
+    const searchMatchForLocale = search.match(searchMatcherForLocale)
+
+    // the value can only either in the path or search query param, there will be no overridden
+    const siteRef = pathMatch?.groups.site || searchMatchForSite?.groups.site
+
+    const localeRef = pathMatch?.groups.locale || searchMatchForLocale?.groups.locale
+    return {siteRef, localeRef}
+}
+
+/**
+ * This function returns the url config from the current configuration
+ * @return {object} - url config
+ */
+export const getUrlConfig = () => {
+    const {app} = getConfig()
+    if (!app.url) {
+        throw new Error("Can't find any valid url config. Please check your configuration file.")
+    }
+    return app.url
+}
+
+/**
+ * This function return the regex for matching site and locale
+ * @param config
+ * @return {{searchMatcherForSite: RegExp, searchMatcherForLocale: RegExp, pathMatcher: RegExp}}
+ */
+export const getConfigMatcher = (config) => {
+    if (!config) {
+        throw new Error('Config is not defined.')
+    }
+
+    const allSites = getSites()
+    const siteIds = []
+    const siteAliases = []
+    const localesIds = []
+    const localeAliases = []
+    allSites.forEach((site) => {
+        siteAliases.push(site.alias)
+        siteIds.push(site.id)
+        const {l10n} = site
+        l10n.supportedLocales.forEach((locale) => {
+            localesIds.push(locale.id)
+            localeAliases.push(locale.alias)
+        })
+    })
+    const sites = [...siteIds, ...siteAliases].filter(Boolean)
+    const locales = [...localesIds, ...localeAliases].filter(Boolean)
+
+    // prettier-ignore
+    // eslint-disable-next-line
+    const searchPatternForSite = `site=(?<site>${sites.join('|')})`
+    // prettier-ignore
+    // eslint-disable-next-line
+    const pathPattern = `(?:\/(?<site>${sites.join('|')}))*(?:\/(?<locale>${locales.join("|")}))*(?!\\w)`
```

Comment:
This is cool.. I didn't know you can have named captured groups like this 👍 And the browser support for it is widespread too.

Reply:
```suggestion
    const pathPattern = `(?:\/(?<site>${sites.join('|')}))?(?:\/(?<locale>${locales.join("|")}))?(?!\\w)`
```
I think what we want is to match the site 0 or 1 time and the locale 0 or 1 time, is that right?

Reply:
yes we want to match it 0 or 1 time, I believev the * is for 0 or more while the ? is 1 or more. There are cases where the site or locale won't be in the path like `/en-GB/women`. The * makes more sense in this case

Reply:
@alexvuong `?` is for 0 or 1 time, while `+` for 1 or more: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Quantifiers#types

Let's use `?` then. It's more strict and accurate than `*`.

Reply:
oh I see, that makes sense. Thank you!